azkaban-common/src/test/java/azkaban/executor/MockExecutorLoader.java

```diff
@@ -54,7 +54,8 @@
   @Override
   public void uploadExecutableFlow(ExecutableFlow flow)
       throws ExecutorManagerException {
-    flows.put(flow.getExecutionId(), flow);
+    ExecutableFlow exFlow = ExecutableFlow.createExecutableFlowFromObject(flow.toObject());
```

Comment:
> Is this an indication of a real production bug?

No! In production code nothing tangles with the job statuses. It's all done in a designated thread  per job.